Matching Frontend-Backend Marriage

nickfong opened the pull request:
In lieu of doing two code reviews on frontend/backend of matching, consolidating into one PR.  Iff I can get this to pass CI before the backend PR is code reviewed, I'll close out #47 and #65.

smartfuse committed:
Matching algorithm and tests. Call /api/v1/matches?id=<userid> to get a list of matches

Commit:
Add request_match function

Commit:
Add framework for matching methods

Commit:
Implement framework for other matching methods

Commit:
Implement hacky version of can_match

kuu2 committed:
Added initial matching bar html and css

kuu2 committed:
Added basic functionality/style to matching requests html

Commit:
Add comments in HTML for controller calls

Commit:
Add pending step definitions for matching

Commit:
Add ng-repeat to sidebar's HTML, update TODOs

Commit:
Add polling to sidebar controller

Commit:
Delete from inboundRequests in accept and decline

kuu2 committed:
Changed wording for matching bar, more changes to popover

smartfuse committed:
Working state for matching

Conflicts:
	spec/models/user_spec.rb

smartfuse committed:
Fixed RSpec before each

Commit:
Stub out API calls for development

kuu2 committed:
Finished most of html

Commit:
Delete the random 's' file

Commit:
Change div to table, add status query @ page load

whsieh committed:
Changed matching API to return an appropriate error message.

kuu2 committed:
Added show/hide attributes to matching container elements, fixed popover

whsieh committed:
Added Invite table migration.

kuu2 committed:
Finished implementing most html views with basic styling

Commit:
Add headers to outbound table

Commit:
Change the class of divs that cause CI failures

Commit:
Add AppState to sidebar controller

Commit:
Add API endpoints to api

Commit:
Implement API calls in controller

Commit:
Update error handling in controller

smartfuse committed:
Invites

smartfuse committed:
Fixed a RSpec test

Commit:
Linking backend and frontend of matching

whsieh committed:
Attempt at getting sidebar jasmine tests to pass.

Commit:
Updating matching cucumber

Change feature file and HTML to match

whsieh committed:
Quick fix to make heroku happy.

Fong- commented on features/step_definitions/matching.rb:
```diff
@@ -0,0 +1,16 @@
+Given(/^that the algorithm matches me with "(.*?)", "(.*?)", "(.*?)", "(.*?)", and "(.*?)"$/) do |match1, match2, match3, match4, match5|
+  pending
```
@smartfuse can you tell me how I should force `matches` to return a preset list of users?

nickfong replied:
@smartfuse can you tell me how I should force `matches` to return a preset list of users?

Fong- replied:
@smartfuse can you tell me how I should force `matches` to return a preset list of users?

Commit:
Remove false error reporting for polling endpoints

invites_controller was returning an error when incoming_requests() and
sent_requests() were polled and the user had no incoming_requests or
sent_requests.  However, this should not be an error since the Angular
controller must continually poll these two endpoints whether or not the
user has any incoming_requests or sent_requests.

This commit also includes removal of trailing whitespace in user.rb.

whsieh committed:
Added remaining conversation page and logout features.

Commit:
Basic sidebar conversations feature.

Fong- commented:
Closing to base off of `iteration 4`
